test/Microsoft.Framework.PackageManager.FunctionalTests/DnuBundleTests.cs

```diff
@@ -1256,5 +1254,111 @@ public void BundleWithNoSourceOptionUpdatesLockFileOnClr(string flavor, string o
                     File.ReadAllText(outputLockFilePath));
             }
         }
+
+
+        [Theory]
+        [MemberData("RuntimeComponents")]
+        public void BundleExcludeWithLongPath(string flavor, string os, string architecture)
+        {
+            var runtimeHomeDir = TestUtils.GetRuntimeHomeDir(flavor, os, architecture);
+
+            var projectStructure = @"{
+  '.': ['project.json', 'Config.json', 'Program.cs'],
+  'Data': {
+    'Input': ['data1.dat', 'data2.dat'],
+    'Backup': ['backup1.dat', 'backup2.dat']
+  },
+  'packages': {}
+}";
+            var expectedOutputStructure = @"{
+  'approot': {
+    'global.json': '',
+    'src': {
+      'PROJECT_NAME': {
+        '.': ['project.json', 'project.lock.json', 'Config.json', 'Program.cs'],
+          'Data': {
+            'Input': ['data1.dat', 'data2.dat']
+          }
+        }
+      }
+    }
+  }".Replace("PROJECT_NAME", _projectName);
+
+            using (var testEnv = new DnuTestEnvironment(runtimeHomeDir, _projectName, _outputDirName))
+            {
+                DirTree.CreateFromJson(projectStructure)
+                    .WithFileContents("project.json", @"{
+  ""bundleExclude"": ""Data/Backup/**"",
+  ""exclude"": ""node_modules""
+}")
+                    .WriteTo(testEnv.ProjectPath);
+
+                BuildLongPath(Path.Combine(testEnv.ProjectPath, "node_modules"));
+
+                var environment = new Dictionary<string, string>()
+                {
+                    { EnvironmentNames.Packages, Path.Combine(testEnv.ProjectPath, "packages") }
+                };
+
+                var exitCode = DnuTestUtils.ExecDnu(
+                    runtimeHomeDir,
+                    subcommand: "bundle",
+                    arguments: string.Format("--out {0}",
+                        testEnv.BundleOutputDirPath),
+                    environment: environment,
+                    workingDir: testEnv.ProjectPath);
+                Assert.Equal(0, exitCode);
+
+                var expectedOutputDir = DirTree.CreateFromJson(expectedOutputStructure)
+                    .WithFileContents(Path.Combine("approot", "src", testEnv.ProjectName, "project.json"), @"{
+  ""bundleExclude"": ""Data/Backup/**"",
+  ""exclude"": ""node_modules""
+}")
+                    .WithFileContents(Path.Combine("approot", "src", testEnv.ProjectName, "project.lock.json"),
+                        BasicLockFile)
+                    .WithFileContents(Path.Combine("approot", "global.json"), @"{
+  ""packages"": ""packages""
+}");
+                Assert.True(expectedOutputDir.MatchDirectoryOnDisk(testEnv.BundleOutputDirPath,
+                    compareFileContents: true));
+            }
+        }
+
+        private string BuildLongPath(string baseDir)
```

Comment:
And of course you can always navigate to the folder via UNC, i.e. `\\localhost\c$\some\long\path\folder` and do whatever operations you want (UNC paths don't have the same short MAX_PATH limit).